command/ca/provisionerbeta/update.go

```diff
@@ -0,0 +1,764 @@
+
+	switch p.Type {
+	case linkedca.Provisioner_JWK:
+		err = updateJWKDetails(ctx, p)
+	case linkedca.Provisioner_ACME:
+		err = updateACMEDetails(ctx, p)
+	case linkedca.Provisioner_SSHPOP:
+		err = updateSSHPOPDetails(ctx, p)
+	case linkedca.Provisioner_X5C:
+		err = updateX5CDetails(ctx, p)
+	case linkedca.Provisioner_K8SSA:
+		err = updateK8SSADetails(ctx, p)
+	case linkedca.Provisioner_OIDC:
+		err = updateOIDCDetails(ctx, p)
+	case linkedca.Provisioner_AWS:
+		err = updateAWSDetails(ctx, p)
+	case linkedca.Provisioner_AZURE:
+		err = updateAzureDetails(ctx, p)
+	case linkedca.Provisioner_GCP:
+		err = updateGCPDetails(ctx, p)
+	// TODO add SCEP provisioner support.
+	default:
+		return fmt.Errorf("unsupported provisioner type %s", p.Type.String())
+	}
+	if err != nil {
+		return err
+	}
+
+	if err = client.UpdateProvisioner(name, p); err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+	b, err := protojson.Marshal(p)
+	if err != nil {
+		return err
+	}
+	if err := json.Indent(&buf, b, "", "  "); err != nil {
+		return err
+	}
+	fmt.Println(buf.String())
+
+	return nil
+}
+
+func updateTemplates(ctx *cli.Context, p *linkedca.Provisioner) error {
+	// Read x509 template if passed
+	if p.X509Template == nil {
+		p.X509Template = &linkedca.Template{}
+	}
+	if x509TemplateFile := ctx.String("x509-template"); ctx.IsSet("x509-template") {
+		if x509TemplateFile == "" {
+			p.X509Template.Template = nil
+		} else {
+			b, err := utils.ReadFile(x509TemplateFile)
+			if err != nil {
+				return err
+			}
+			p.X509Template.Template = b
+		}
+	}
+	if x509TemplateDataFile := ctx.String("x509-template-data"); ctx.IsSet("x509-template-data") {
+		if x509TemplateDataFile == "" {
+			p.X509Template.Data = nil
+		} else {
+			b, err := utils.ReadFile(x509TemplateDataFile)
+			if err != nil {
+				return err
+			}
+			p.X509Template.Data = b
+		}
+	}
+	// Read ssh template if passed
+	if p.SshTemplate == nil {
+		p.SshTemplate = &linkedca.Template{}
+	}
+	if sshTemplateFile := ctx.String("ssh-template"); ctx.IsSet("ssh-template") {
+		if sshTemplateFile == "" {
+			p.SshTemplate.Template = nil
+		} else {
+			b, err := utils.ReadFile(sshTemplateFile)
+			if err != nil {
+				return err
+			}
+			p.SshTemplate.Template = b
+		}
+	}
+	if sshTemplateDataFile := ctx.String("ssh-template-data"); ctx.IsSet("ssh-template-data") {
+		if sshTemplateDataFile == "" {
+			p.SshTemplate.Data = nil
+		} else {
+			b, err := utils.ReadFile(sshTemplateDataFile)
+			if err != nil {
+				return err
+			}
+			p.SshTemplate.Data = b
+		}
+	}
+	return nil
+}
+
+func updateClaims(ctx *cli.Context, p *linkedca.Provisioner) {
+	if p.Claims == nil {
+		p.Claims = &linkedca.Claims{}
+	}
+	if ctx.IsSet("disable-renewal") {
+		p.Claims.DisableRenewal = ctx.Bool("disable-renewal")
+	}
+	xc := p.Claims.X509
+	if xc == nil {
+		xc = &linkedca.X509Claims{}
+	}
+	if ctx.IsSet("x509") {
+		xc.Enabled = ctx.Bool("x509")
+	}
+	d := xc.Durations
+	if d == nil {
+		d = &linkedca.Durations{}
```

Comment:
This new printer won't set xc.Durations.

Reply:
Can you give this another look and make sure I've resolved these correctly?

Reply:
Without testing it, it looks ok now.